H2TauTau/python/heppy/analyzers/METAnalyzer.py

```diff
@@ -8,12 +8,23 @@
 from PhysicsTools.Heppy.analyzers.core.Analyzer import Analyzer
 from PhysicsTools.Heppy.analyzers.core.AutoHandle import AutoHandle
 
+from PhysicsTools.HeppyCore.utils.deltar import cleanObjectCollection
+
 gSystem.Load("libCMGToolsH2TauTau")
 
 from ROOT import HTTRecoilCorrector as RC
 
 LorentzVector = ROOT.Math.LorentzVector(ROOT.Math.PxPyPzE4D("double"))
 
+def get_final_ptcs(ptc):
+    if ptc.numberOfDaughters() == 0 :
+        return [ptc]
+    else :
+        final_ptcs = []
+        for N in range(ptc.numberOfDaughters()):
```

Comment:
avoid single letter variables, except for i, j in enumeration. 
don't use single capital variables. 
here ndau would be good